tape/src/main/java/com/squareup/tape/FileObjectQueue.java

```diff
@@ -102,6 +101,19 @@ public FileObjectQueue(File file, Converter<T> converter) throws IOException {
     }
   }
 
+  public final void remove(int n) throws IOException {
+    try {
+      queueFile.remove(n);
+      if (listener != null) {
+        for (int i = 0; i < n; i++) {
+          listener.onRemove(this);
+        }
+      }
+    } catch (IOException e) {
+      throw new FileException("Failed to remove.", e, file);
```

Comment:
This is the worst thing I've ever written in my life. It's not entirely my fault that it's in here, but I blame me for allowing it to remain when we yanked this crap out of the old, old Retrofit project.